src/overlays/actors/ovl_Bg_Spot18_Basket/z_bg_spot18_basket.c

```diff
@@ -21,39 +37,397 @@ const ActorInit Bg_Spot18_Basket_InitVars = {
+        osSyncPrintf("Ｅｒｒｏｒ : 変化壷蓋発生失敗(%s %d)\n", "../z_bg_spot18_basket.c", 351);
+        osSyncPrintf(VT_RST);
+        Actor_Kill(actor);
+    }
+}
+
+void BgSpot18Basket_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot18Basket* this = THIS;
+
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+    Collider_DestroyJntSph(globalCtx, &this->colliderJntSph);
+}
+
+void func_808B7AEC(BgSpot18Basket* this) {
+    this->actionFunc = func_808B7AFC;
+}
+
+void func_808B7AFC(BgSpot18Basket* this, GlobalContext* globalCtx) {
+    if (Flags_GetSwitch(globalCtx, (this->dyna.actor.params >> 8) & 0x3F)) {
+        func_800800F8(globalCtx, 4220, 80, this, 0);
+        func_808B7B58(this);
+    }
+}
+
+void func_808B7B58(BgSpot18Basket* this) {
+    this->actionFunc = func_808B7B6C;
+    this->unk_216 = 0;
+}
+
+void func_808B7B6C(BgSpot18Basket* this, GlobalContext* globalCtx) {
+    if (this->unk_216 > 20) {
+        func_808B7BB0(this);
+        this->dyna.actor.child->parent = NULL;
+        this->dyna.actor.child = NULL;
+    }
+}
+
+void func_808B7BB0(BgSpot18Basket* this) {
+    this->actionFunc = func_808B7BCC;
+    this->unk_210 = this->unk_20C = 0;
+}
+
+void func_808B7BCC(BgSpot18Basket* this, GlobalContext* globalCtx) {
+    f32 positionDiff;
+    Actor* colliderBaseAc;
+
+    Math_ApproxS(&this->unk_210, 500, 30);
+
+    this->dyna.actor.shape.rot.y += this->unk_210;
+
+    Math_ApproxF(&this->unk_208, 50.0f, 1.5f);
+    Math_ApproxS(&this->unk_20C, 400, 15);
+
+    this->unk_20E += this->unk_20C;
+
+    this->dyna.actor.posRot.pos.x = (Math_Sins(this->unk_20E) * this->unk_208) + this->dyna.actor.initPosRot.pos.x;
+    this->dyna.actor.posRot.pos.z = (Math_Coss(this->unk_20E) * this->unk_208) + this->dyna.actor.initPosRot.pos.z;
+
+    if (this->colliderJntSph.base.acFlags & 2) {
+        colliderBaseAc = this->colliderJntSph.base.ac;
+
+        if (colliderBaseAc != NULL) {
+            positionDiff = colliderBaseAc->posRot.pos.y - this->dyna.actor.posRot.pos.y;
+
+            if (positionDiff > 120.0f && positionDiff < 200.0f) {
+                if (Math3D_Dist2DSq(colliderBaseAc->posRot.pos.z, this->colliderJntSph.base.ac->posRot.pos.x,
+                                    this->dyna.actor.posRot.pos.z, this->dyna.actor.posRot.pos.x) < SQ(32.0f)) {
+                    func_800800F8(globalCtx, 4210, 240, this, 0);
+                    func_808B7D38(this);
+                    func_8003EBF8(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+                }
+            }
+        }
+    }
+    func_8002F974(this, NA_SE_EV_ELEVATOR_MOVE - SFX_FLAG);
+}
+
+void func_808B7D38(BgSpot18Basket* this) {
+    this->actionFunc = func_808B7D50;
+    this->unk_216 = 0;
+    this->unk_214 = 0;
+}
+
+void func_808B7D50(BgSpot18Basket* this, GlobalContext* globalCtx) {
+    f32 tempValue2;
+    f32 tempValue;
+
+    if (this->unk_216 > 120) {
+        Math_ApproxS(&this->unk_210, 1000, 50);
```

Comment:
rotation values, use hex! (`this->unk_210` is added to `rot.y` 5 lines below)

also in many other places in this file